include/trompeloeil.hpp

```diff
@@ -1498,9 +1498,13 @@ template <typename T>
 
   template <typename T, typename Disposer>
   class list<T, Disposer>::iterator
-    : public std::iterator<std::bidirectional_iterator_tag, T>
   {
     friend class list<T, Disposer>;
+    typedef std::bidirectional_iterator_tag iterator_category;
```

Comment:
Nitpick: In the original code these types are declared `public`.
Now they are declared `private`.  I suggest they remain `public`.